singa_auto/admin/admin.py

```diff
@@ -141,17 +141,108 @@ def ban_user(self, email):
     ####################################
 
     def create_dataset(self, user_id, name, task, data_file_path):
+        """ This function is to return statistics to frontend during dataset uploading process only, which is different from the dataset-reading methods during model training """
+        # Store dataset in data folder
+        print('begin saving to local path')
+        store_dataset = self._data_store.save(data_file_path)
+        # Get metadata for dataset. 'store_dataset' is a dictionary contains the following info only
+        store_dataset_id = store_dataset.id
+        size_bytes = store_dataset.size_bytes
+        stat = dict()
+        if len(os.path.splitext(data_file_path)) == 2 and os.path.splitext(
```

Comment:
pls add comments for this code block.
what is the logic here for the if else ...

Reply:
Comments added to newest pr update